clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/FailedReconnectStoreProxy.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.internal.store;
+
+import org.ehcache.clustered.client.internal.store.lock.LockingServerStoreProxy;
+import org.ehcache.clustered.common.internal.store.Chain;
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.concurrent.TimeoutException;
+
+public class FailedReconnectStoreProxy implements LockingServerStoreProxy {
+  private final Throwable failure;
+  private final String cacheId;
+
+  public FailedReconnectStoreProxy(String cacheId, Throwable failure) {
+    this.cacheId = cacheId;
+    this.failure = failure;
+  }
+
+  @Override
+  public ChainEntry get(long key) {
+    throw new RuntimeException("Cache " + getCacheId() + " failed reconnecting to cluster", failure);
```

Comment:
shouldn't they be `IllegalState` instead ? Which also would match closer the underlying exception in case we rename it to something else than Perpetual...

Reply:
I think you're right that 'IllegalStateException' or maybe something new and Ehcache specific make sense here. I disagree on renaming the perpetual exception though. ;)